sycl/doc/EnvironmentVariables.md

```diff
@@ -84,9 +84,22 @@ The following examples further illustrate the usage of this environment variable
 Notes:
 - The backend argument is always required. An error will be thrown if it is absent.
 - Additionally, the backend MUST be followed by colon ( `:` ) and at least one device specifier of some sort, else an error is thrown.
-- For sub-devices and sub-sub-devices, the parent device must support partitioning (`info::partition_property::partition_by_affinity_domain` and `info::partition_affinity_domain::next_partitionable`. See the SYCL 2020 specification for a precise definition.) For Intel GPUs, the sub-device and sub-sub-device syntax can be used to expose tiles or CCSs to the SYCL application as root devices.  The exact mapping between sub-device, sub-sub-device, tiles, and CCSs is specific to the hardware.
+- The sub-device and sub-sub-device syntax attempt to partition the root device
+  according to the rules defined by
+  `info::partition_property::partition_by_affinity_domain` and
+  `info::partition_affinity_domain::next_partitionable`.
+  (See the SYCL 2020 specification for a precise definition.)
+  The root device is determined by the underlying backend.
+- When using the Level Zero backend, see also the documentation of the
+  [`ZE_FLAT_DEVICE_HIERARCHY`][ze-env] environment variable because it affects
+  how this backend exposes root devices to SYCL.
+  For Intel GPUs, the sub-device and sub-sub-device syntax can be used to
+  expose tiles or CCSs to the SYCL application as SYCL root devices, however
```

Comment:
According to 
https://www.intel.com/content/www/us/en/docs/oneapi/optimization-guide-gpu/2023-0/intel-iris-xe-gpu-architecture.html#XE-TERMINOLOGY-CHANGES we should use "stack" instead of "tile". I think there might be other instances where we use old terminology, so it makes sense to make this update separately and apply to all documents in the repository.